pkg/kv/kvserver/replica_proposal_buf.go

```diff
@@ -888,6 +889,8 @@ func proposeBatch(
 		//      slice of entries. See etcd-io/raft#57.
 		maybeDeductFlowTokens(ctx, p.flowControlHandle(ctx), handles, ents)
 	}
+
+	p.maybeRegister(ctx, ents)
```

Comment:
Here we know both the `ents` slice, and the corresponding `props` slice which is parallel to `ents`. So we can avoid the `proposals` map lookup inside the `maybeRegister` call.

Reply:
Changed. Thanks for the insight, I didn't realize we had both here.